databricks_cli/pipelines/cli.py

```diff
@@ -49,27 +50,144 @@
+@click.option('--settings', default=None, type=PipelineSettingClickType(),
+              help=PipelineSettingClickType.help)
+@click.option('--pipeline-id', default=None, type=PipelineIdClickType(),
+              help=PipelineIdClickType.help)
+@click.option('--allow-duplicate-names', is_flag=True,
+              help="Skip duplicate name check while editing pipeline.")
+@debug_option
+@profile_option
+@pipelines_exception_eater
+@provide_api_client
+def edit_cli(api_client, settings_arg, settings, pipeline_id, allow_duplicate_names):
+    """
+    Edits a pipeline according to the pipeline settings. The pipeline settings are a
+    JSON document that defines a Delta Live Tables pipeline on Databricks.
+
+    If another pipeline with the same name exists, pipeline settings will not be edited.
+    This check can be disabled by adding the --allow-duplicate-names option.
+
+    Note that if an ID is both specified in the settings and passed via --pipeline-id,
+    the two ids must be the same, or the command will fail.
+
+    Usage:
+
+    databricks pipelines edit example.json
+
+    OR
+
+    databricks pipelines edit --settings example.json
+    """
+    if bool(settings_arg) == bool(settings):
+        raise ValueError('Settings should be provided either as an argument ' +
+                         '(Eg: databricks pipelines edit example.json) or as ' +
+                         'an option (Eg: databricks pipelines edit --settings example.json).')
+
+    src = settings_arg if bool(settings_arg) else settings
+    settings_obj = _read_settings(src)
+    settings_dir = os.path.dirname(src)
+
+    if (pipeline_id and 'id' in settings_obj) and pipeline_id != settings_obj["id"]:
+        raise ValueError(
+            "The ID provided in --pipeline_id '{}' is different from the ID provided "
+            "in the settings '{}'. Resolve the conflict and try the command again. ".format(
+                pipeline_id, settings_obj["id"])
+        )
+
+    settings_obj['id'] = pipeline_id or settings_obj.get('id', None)
+    _validate_pipeline_id(settings_obj['id'])
+
+    try:
+        PipelinesApi(api_client).edit(settings_obj, settings_dir, allow_duplicate_names)
+    except requests.exceptions.HTTPError as e:
+        _handle_duplicate_name_exception(settings_obj, e, is_create_pipeline=False)
+    click.echo("Successfully edited pipeline settings: {}".format(
+        _get_pipeline_url(api_client, settings_obj['id'])))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='[Deprecated] This command is deprecated, use create and edit '
+                          'commands instead.\n Deploys a pipeline according to the '
+                          'pipeline settings.')
+@click.argument('settings_arg', default=None, required=False)
+@click.option('--settings', default=None, type=PipelineSettingClickType(),
+              help=PipelineSettingClickType.help)
+@click.option('--spec', default=None, type=PipelineSpecClickType(),
```

Comment:
Note: I am marking --spec as deprecated but user can continue to use --spec (since the deploy is deprecated, we can eventually get rid of the entire command). If user switches to --settings that should work as well but only one of them is expected to be passed in as input.